airbyte-webapp/src/components/TreeView/components/SyncSettingsCell.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useMemo, useState } from "react";
+import styled from "styled-components";
+
+import { Cell } from "../../SimpleTableComponents";
+import DropDown from "../../DropDown";
+import { IDataItem } from "../../DropDown/components/ListItem";
+import { SyncSchemaStream } from "../../../core/resources/Schema";
+import { useIntl } from "react-intl";
+
+const DropDownContainer = styled.div`
+  padding-right: 10px;
+`;
+
+const StyledDropDown = styled(DropDown)`
+  & ~ .rw-popup-container {
+    min-width: 260px;
+    left: auto;
+  }
+`;
+
+type IProps = {
+  item: SyncSchemaStream;
+  onSelect: (data: IDataItem) => void;
+};
+
+const SyncSettingsCell: React.FC<IProps> = ({ item, onSelect }) => {
+  const formatMessage = useIntl().formatMessage;
+  const [supportIncremental, setSupportIncremental] = useState(false);
+
+  const data: IDataItem[] = useMemo(
+    () =>
+      item.supportedSyncModes
+        .filter(mode => {
+          if (mode === "incremental") {
+            setSupportIncremental(true);
+            return false;
+          }
+          return true;
+        })
+        .map(mode => ({
+          value: mode,
+          text: formatMessage({
+            id: `sources.${mode}`,
+            defaultMessage: mode
+          })
+        })),
+    [formatMessage, item.supportedSyncModes]
+  );
+
+  const fullData = useMemo(() => {
+    // If INCREMENTAL is included in the supported sync modes...
+    if (supportIncremental) {
+      if (item.sourceDefinedCursor) {
+        // If sourceDefinedCursor is true, In the dropdown we should just have one row for incremental
+        data.push({
+          text: formatMessage({
+            id: "sources.incrementalSourceCursor"
+          }),
+          value: "incremental"
+        });
+      } else {
+        // If sourceDefinedCursor is false...
+
+        // If defaultCursorField is set, then the field specified in there should be at the top of the list
+        // and have the word "(default)" next to it
+        item.defaultCursorField.forEach(field =>
```

Comment:
i did not make this clear in the issue. `defaultCursorField` and `cursorField` is an array of strings, because in the future we plan to support using a nested field as the cursor. e.g. if the object were `{ name: "blah", metadata: { updated_at: "2020-11-01" } }` then the cursor field might be `["metdata", "updated_at"]`. 

That being said that is for future compatibility. It is safe to assume right now is that if the cursor field is set then it is an array with a single element in it. We can worry about nested cursor fields later.

so for this line of code for example, it would pretty much be something like this instead of the map:
```
if(item.defaultCursorField != null) {
          data.push({
            text: formatMessage(
              {
                id: "sources.incrementalDefault"
              },
              { value: field }
            ),
            value: item.defaultCursorField[0] ,
            secondary: true,
            groupValue: "incremental",
            groupValueText: formatMessage({
              id: "sources.incremental"
            })
          })
        );
}
```

sorry for the confusion on that! hope it clears it up.